src/main/host/syscall_types.rs

```diff
@@ -90,6 +90,18 @@ impl From<SysCallReg> for u64 {
     }
 }
 
+impl From<u32> for SysCallReg {
+    fn from(v: u32) -> Self {
+        Self { as_u64: v as u64 }
+    }
+}
+
+impl From<SysCallReg> for u32 {
+    fn from(v: SysCallReg) -> u32 {
+        (unsafe { v.as_u64 }) as u32
```

Comment:
This should probably be a  TryFrom...?

Reply:
I think so, but the existing conversions from other types like `i32` already use `From` rather than `TryFrom`, so I just made it consistent with the others for now. I'll make a note to fix all of these in a future PR.